fr-dev

```diff
@@ -3,51 +3,82 @@
 set -e
 set -x
 
+function clean {
+    docker-compose stop
+    docker-compose rm -f
+}
+
+function rundb {
+    docker-compose exec db $@
+}
+
+function runweb {
+    docker-compose run --rm web "$@"
+    clean
+}
+
+function initdb {
+    runweb python manage.py migrate
+    runweb python manage.py initdb
+}
+
 case "$1" in
     up)
         docker-compose up --build
         ;;
     run)
-        docker-compose run --service-ports web
+        docker-compose run --rm --service-ports web
         ;;
     initdb)
-        docker-compose exec web python manage.py migrate
-        docker-compose exec web python manage.py shell -c 'import init_db'
+        initdb
         ;;
     resetdb)
-        docker-compose stop web
-        docker-compose exec db dropdb -U fedireads fedireads
-        docker-compose exec db createdb -U fedireads fedireads
-        docker-compose start web
-        docker-compose exec web python manage.py migrate
-        docker-compose exec web python manage.py shell -c 'import init_db'
+        clean
+        docker-compose up --build -d
+        rundb dropdb -U fedireads fedireads
+        rundb createdb -U fedireads fedireads
+        initdb
+        clean
         ;;
     makemigrations)
-        docker-compose exec web python manage.py makemigrations
+        runweb python manage.py makemigrations
         ;;
     migrate)
-        docker-compose exec web python manage.py migrate
+        runweb python manage.py migrate
+        ;;
+    bash)
+        runweb bash
         ;;
     shell)
-        docker-compose exec web python manage.py shell
+        runweb python manage.py shell
```

Comment:
when I have docker-compose running, and then in another window open the shell using `./fr-dev shell`, when I exit the shell it exits all the docker images as well in the other window. I think commands like this need to keep exec so that they don't get tangled up with the main process running the images?

Reply:
I think this goes for makemigrations, migrate, bash, and shell.

Reply:
(this applies to all the existing fr-dev commands)

Reply:
That's a great callout! Yeah, the "run" paired with clean recipe would be for one-off things where it would otherwise be an extra step to spin up the containers, then execute the command in another tab. If you're typically always running the application when you run these commands, then exec makes a lot more sense!

Are there any of these fr-dev commands that you want to be able to run independently? Like initializing or resetting the db, or maybe running a test suite on CI?

Either way for convenience I reverted these all back to use exec, and leaved the run helper that cleans up after itself in there in case it's useful in 061193c.